improver/nbhood/square_kernel.py

```diff
@@ -481,13 +475,13 @@ def run(self, cube, radius, mask_cube=None):
         """
         # If the data is masked, the mask will be processed as well as the
         # original_data * mask array.
+        check_radius_against_distance(cube, radius)
         original_attributes = cube.attributes
         original_methods = cube.cell_methods
-        grid_cells_x = (
-            convert_distance_into_number_of_grid_cells(
-                cube, radius,
-                max_distance_in_grid_cells=MAX_RADIUS_IN_GRID_CELLS))
-        grid_cells_y = grid_cells_x
+        grid_cells = (
+            distance_to_number_of_grid_cells(cube, radius,
+                                             MAX_RADIUS_IN_GRID_CELLS))
```

Comment:
As in `circular_kernel`, you no longer need `MAX_RADIUS_IN_GRID_CELLS` in these calls (as you're going to be checking this aspect in the `check_radius_against_distance` method).